exercises/phone-number/src/example.c

```diff
@@ -0,0 +1,70 @@
+#include "phone_number.h"
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+
+#define AREA_CODE_LENGTH         (3)
+#define EXCHANGE_LENGTH          (3)
+#define EXTENSION_LENGTH         (4)
+#define VALID_NUMBER_LENGTH      AREA_CODE_LENGTH + EXCHANGE_LENGTH + EXTENSION_LENGTH
+#define FORMATTED_LENGTH         VALID_NUMBER_LENGTH + 4
+
+#define INVALID_NUMBER_RESULT    "0000000000"
+
+static void remove_leading_digit(char *phone_number) {
+   char *temp = calloc(VALID_NUMBER_LENGTH, sizeof(char));
+   strcpy(temp, &phone_number[1]);
+   strcpy(phone_number, temp);
+   free(temp);
+}
+
+char *phone_number_clean(const char *input)
+{
+   char *output = calloc(VALID_NUMBER_LENGTH + 1, sizeof(char));
+
+   for (size_t i = 0; i < strlen(input); i++) {
+      if (isdigit(input[i])) {
+         strncat(output, &input[i], 1);
+      }
+   }
+
+   if ((strlen(output) > 11) || (strlen(output) < 10)) {
+      strcpy(output, INVALID_NUMBER_RESULT);
+   } else if (strlen(output) == 11) {
+      if (output[0] == '1') {
+         remove_leading_digit(output);
+      } else {
+         strcpy(output, INVALID_NUMBER_RESULT);
+      }
+   }
+
+   return output;
+}
+
+char *phone_number_get_area_code(const char *input)
+{
+   char *output = calloc(AREA_CODE_LENGTH, sizeof(char));
+   strncpy(output, phone_number_clean(input), AREA_CODE_LENGTH);
+
+   return output;
+}
+
+char *phone_number_format(const char *input)
```

Comment:
`phone_number_get_area_code` uses `phone_number_clean` internally, but this doesn't. I think we should add a test that requires this for consistency.